src/environmentd/src/http.rs

```diff
@@ -530,6 +558,12 @@ where
             .await
             .unwrap_or_default();
 
+        let peer_addr = req
+            .extensions
+            .get::<ConnectInfo<SocketAddr>>()
+            .map(|ci| ci.0.ip())
+            .unwrap();
```

Comment:
I know the nearby code sadly doesn't use `expect`, but I do think it's a good bit easier to reason about the safety with `expect`:

```suggestion
            .get::<ConnectInfo<SocketAddr>>()
            .expect("ConnectInfo extension guaranteed to exist")
            .0.ip();
```